src/compiler/checker.ts

```diff
@@ -9257,6 +9316,12 @@ namespace ts {
                         if (!bodyDeclaration) {
                             bodyDeclaration = node;
                         }
+
+                        // abstract functions cannot have an implementation.
+                        // Extra checks are to avoid reporting multiple errors relating to the "abstractness" of the node.
+                        if (currentNodeFlags & NodeFlags.Abstract && node.kind === SyntaxKind.MethodDeclaration && !isConstructor) {

```

Comment:
Constructors cannot be abstract. Consider the following:

```ts
abstract class A {
    abstract constructor();
}
class B extends A {
    constructor() {
        super(); // Calling an abstract method, oops!
    }
}
```

We report an error for the placing the `abstract` keyword on `constructor()` in `checkGrammarModifers()`.